meson/part/exports/copy-export.sh

```diff
@@ -44,6 +44,15 @@ while [ "$#" -gt 0 ]; do
         staticlib)
             ar -t "$2" | xargs ar crs "${target_dir}/lib/$3"
             ;;
+        sharedlib)
+            filename="$(basename "$2")"
+            major_versioned_name="$3"
+            # libfoo.so.5 -> libfoo.so (remove the last part delimited by dot)
+            unversioned_name="$(echo "$major_versioned_name" | rev | cut -d. -f2- | rev)"
```

Comment:
Did you try the parameter expansion notation I suggested?
This could be `unversioned_name="${major_versioned_name%.*}"`